### PR TITLE
skip rendering the table of related objects in the confirmation page.

### LINKED ISSUE #6465: "Edit All" view should skip rendering the table of related objects in the confirmation page
### Proposed Changes
"Edit All" view should skip rendering the table of related objects in the confirmation page, just like Delete all in https://github.com/nautobot/nautobot/pull/6420

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes #6465 
# What's Changed

# Screenshots

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/views/mixins.py [hunk 1]
```diff
@@ -1178,38 +1184,42 @@ def perform_bulk_update(self, request, **kwargs):  # pylint: disable=arguments-d
...
-            messages.warning(
-                request,
-                f"No {queryset.model._meta.verbose_name_plural} were selected to update.",
-            )
-            return redirect(self.get_return_url(request))
-        data.update({"table": table})
+        table = None
+        if not edit_all:
+            table_class = self.get_table_class()
+            table = table_class(queryset.filter(pk__in=self.pk_list), orderable=False)
```
Comment (glennmatthews) [inline nautobot/core/views/mixins.py:1209]: Is the filter() call here redundant with the filter() call at line 1194 above?

File: nautobot/extras/forms/base.py [hunk 2]
```diff
@@ -62,3 +64,9 @@ class NautobotBulkEditForm(
     RelationshipModelBulkEditFormMixin,
 ):
     """Base class for bulk-edit forms for models that support relationships, custom fields and notes."""
```
Comment (glennmatthews) [inline nautobot/extras/forms/base.py:66]: Are there any models that have a BulkEditForm that isn't a a NautobotBulkEditForm subclass?

File: nautobot/core/views/generic.py [hunk 3]
```diff
@@ -1041,7 +1041,8 @@ def post(self, request, **kwargs):
                 try:
                     with deferred_change_logging_for_bulk_operation():
                         updated_objects = []
-                        for obj in self.queryset.filter(pk__in=form.cleaned_data["pk"]):
+                        queryset = queryset if edit_all else queryset.filter(pk__in=form.cleaned_data["pk"])
```
Comment (glennmatthews) [inline nautobot/core/views/generic.py:1044]: Is this filter() redundant with the filter() at line 1024 above?

**Comment by itdependsnetworks** [comment]
Can you take a look here: https://github.com/nautobot/nautobot-app-golden-config/blob/develop/nautobot_golden_config/views.py#L280 and check if there is any impact from this change?

File: nautobot/core/views/mixins.py [hunk 4]
```diff
@@ -912,23 +912,23 @@ def perform_update(self, request, *args, **kwargs):  # pylint: disable=arguments
...
 
         This method handles the retrieval of a queryset of model instances that match the specified
         filter criteria in the request parameters, allowing a bulk delete operation to be performed
         on all matching instances.
         """
         model = self.queryset.model
 
-        # This Mixin is currently been used by both NautobotViewSet ObjectBulkDestroyViewMixin
-        # and BulkDeleteView which uses different keys for accessing filterset
+        # This Mixin is currently been used by both NautobotViewSet ObjectBulkDestroyViewMixin, ObjectBulkUpdateViewMixin
```
Comment (HanlinMiao) [inline nautobot/core/views/mixins.py:930]: Perhaps?
```suggestion
        # This Mixin is currently been used by both NautobotUIViewSet ObjectBulkDestroyViewMixin, ObjectBulkUpdateViewMixin
```

File: nautobot/core/views/renderers.py [hunk 5]
```diff
@@ -249,19 +249,22 @@ def get_context(self, data, accepted_media_type, renderer_context):
...
+                    table = self.construct_table(view, pk_list=pk_list)
             elif view.action == "bulk_create":  # 3.0 TODO: remove, replaced by ImportObjects system Job
                 form = view.get_form()
                 if request.data:
                     table = data.get("table")
             elif view.action == "bulk_update":
+                edit_all = request.POST.get("_all")
                 pk_list = getattr(view, "pk_list", [])
-                if pk_list:
+                if pk_list or edit_all:
```
Comment (HanlinMiao) [inline nautobot/core/views/renderers.py:262]: Can we delete the if statement here?

**Comment by glennmatthews** [comment]
> Can you take a look here: https://github.com/nautobot/nautobot-app-golden-config/blob/develop/nautobot_golden_config/views.py#L280 and check if there is any impact from this change?

Yes, most likely GC will need to be updated to account for the changes from #6420.

File: nautobot/core/templates/generic/object_bulk_update.html [hunk 6]
```diff
@@ -27,14 +27,16 @@ <h1>{% block title %}Editing {{ table.rows|length }} {{ obj_type_plural|betterti
...
         <div class="col-md-8">
             <div class="panel panel-default">
                 <div class="table-responsive">
                     {% render_table table 'inc/table.html' %}
                 </div>
             </div>
         </div>
-        <div class="col-md-4">
+        {% endif %}
+        <div class="{% if edit_all %} col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 {% else %} col-md-4 {% endif %}">
```
Comment (glennmatthews) [inline nautobot/core/templates/generic/object_bulk_update.html:39]: Do we really need a separate `edit_all` flag in this template or could this just be changed to `if not table`?

File: nautobot/core/testing/views.py [hunk 7]
```diff
@@ -1113,14 +1113,11 @@ def test_bulk_edit_objects_with_permission(self):
             self.validate_object_data_after_bulk_edit(pk_list)
 
         @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
-        def test_bulk_edit_form_contains_all_pks(self):
-            # We are testing the intermediary step of bulk_edit with pagination applied.
+        def test_bulk_edit_form_contains_all_objects(self):
```
Comment (glennmatthews) [inline nautobot/core/testing/views.py:1116]: Renaming this test case *might* be impactful to Apps, we should consider keeping the name unchanged.

File: nautobot/core/views/generic.py [hunk 8]
```diff
@@ -1041,7 +1041,7 @@ def post(self, request, **kwargs):
                 try:
                     with deferred_change_logging_for_bulk_operation():
                         updated_objects = []
-                        for obj in self.queryset.filter(pk__in=form.cleaned_data["pk"]):
+                        for obj in queryset:
```
Comment (glennmatthews) [inline nautobot/core/views/generic.py:1044]: Any significance to the fact that we were doing `pk__in=form.cleaned_data["pk"]` before but now (lines 1023-1024 above) we're doing `pk__in=request.POST.getlist("pk")`? Any subtle behavior change that results from this?

File: nautobot/core/views/renderers.py (hunk 5 above)
Comment (glennmatthews) [reply nautobot/core/views/renderers.py:262]: Not sure - don't we need to account for the possibility of an empty `pk_list`?

Comment (HanlinMiao) [reply nautobot/core/views/renderers.py:262]: I see we are accounting for the scenario where no objects are selected at all. That makes sense.

File: nautobot/core/views/generic.py (hunk 8 above)
Comment (glennmatthews) [reply nautobot/core/views/generic.py:1044]: My concern was about the non-`edit_all` case - just trying to understand the significance of the differing logic between line 1044 (before) and line 1024 (after).

File: nautobot/core/views/generic.py [hunk 9]
```diff
@@ -1041,7 +1041,8 @@ def post(self, request, **kwargs):
                 try:
                     with deferred_change_logging_for_bulk_operation():
                         updated_objects = []
-                        for obj in self.queryset.filter(pk__in=form.cleaned_data["pk"]):
+                        queryset = queryset if edit_all else self.queryset.filter(pk__in=form.cleaned_data["pk"])
```
Comment (glennmatthews) [inline nautobot/core/views/generic.py:1044]: In the non-edit-all case, doesn't this overwrite the assignment at line 1024? Should it perhaps be:

```suggestion
                        queryset = queryset if edit_all else queryset.filter(pk__in=form.cleaned_data["pk"])
```

Or alternately, is the assignment at line 1024 unnecessary/irrelevant?

Comment (timizuoebideri1) [reply nautobot/core/views/generic.py:1044]: `queryset` is utilised on line 1140

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**